integration/create_test.go

```diff
@@ -190,46 +152,62 @@ func TestSociCreate(t *testing.T) {
 				t.Fatalf("cannot get soci index: %v", err)
 			}
 
-			if sociIndex.MediaType != ocispec.MediaTypeArtifactManifest {
-				t.Fatalf("unexpected index media type; expected = %v, got = %v", ocispec.MediaTypeArtifactManifest, sociIndex.MediaType)
+			m, err := getManifestDigest(sh, imgInfo.ref, platform)
+			if err != nil {
+				t.Fatalf("failed to get manifest digest: %v", err)
 			}
 
-			if sociIndex.ArtifactType != soci.SociIndexArtifactType {
-				t.Fatalf("unexpected index artifact type; expected = %v, got = %v", soci.SociIndexArtifactType, sociIndex.ArtifactType)
-			}
+			validateSociIndex(t, sh, *sociIndex, m, nil)
+		})
+	}
+}
 
-			expectedAnnotations := map[string]string{
-				soci.IndexAnnotationBuildToolIdentifier: "AWS SOCI CLI v0.1",
-			}
+func validateSociIndex(t *testing.T, sh *shell.Shell, sociIndex soci.Index, imgManifestDigest string, includedLayers map[string]struct{}) {
```

Comment:
nit: can we move this to testutil.go or something?

Reply:
Yeah I think it's good to move these functional helpers to a common place. Maybe we can track this in https://github.com/awslabs/soci-snapshotter/issues/247, and do it two steps (first make changes so it's reusable (in this PR), then we move those helpers to a better place without change). 

I found doing so is easier to review (easier to visualize what changes are made; otherwise it will be deleting a func in one file and adding a slightly changed func in another file).